### PR TITLE
Added support for hash field expiration commands

### PR DESCRIPTION
**Pull request by vladvildanov**
This PR adds new commands operates on hash field expiration functionality

### CODE REVIEW

**Comment by vladvildanov** [comment]
PR still on draft, until:

- Redis unstable docker image will support given functionality
- `HSETF, HGETF` commands will be supported (https://github.com/redis/redis/pull/13209)

File: src/ClientContextInterface.php
Comment (gerzse) [inline src/ClientContextInterface.php:178]: Call `keyValuePairs` `dictionary`, like in `hmset`?

File: src/ClientInterface.php
Comment (gerzse) [inline src/ClientInterface.php:187]: Same here, use same naming like in `hmset`, i.e. `dictionary`?

File: src/Command/Argument/Hash/CommonExpiration.php
Comment (gerzse) [inline src/Command/Argument/Hash/CommonExpiration.php:42]: Being picky here, missing end dot at two sentences. And some more at the other method below.

File: src/Command/Argument/Hash/CommonExpiration.php
Comment (gerzse) [inline src/Command/Argument/Hash/CommonExpiration.php:23]: In my mind these two don't belong together. The {NX,XX,GT,LT} set is for overwriting rules, while {EX,PX,EXAT,PXAT} are basically just parameter name in compact form. For example {NX,XX,GT,LT} applies to other hash expiration commands, while the other set does not.

File: src/Command/Argument/Hash/HGetFArguments.php
Comment (gerzse) [inline src/Command/Argument/Hash/HGetFArguments.php:26]: What if someone first calls `setPersist` and then calls `setTtlModifier`?

File: src/Command/Argument/Hash/CommonExpiration.php
Comment (gerzse) [inline src/Command/Argument/Hash/CommonExpiration.php:60]: Make sure at most one modifier is used.

File: src/Command/Argument/Hash/CommonExpiration.php
Comment (gerzse) [inline src/Command/Argument/Hash/CommonExpiration.php:86]: Here also, at most one modifier can be used from the group.

File: src/Command/Argument/Hash/HSetFArguments.php
Comment (gerzse) [inline src/Command/Argument/Hash/HSetFArguments.php:60]: Check that at most one modifier from the group is used.

File: src/Command/Argument/Hash/HSetFArguments.php
Comment (gerzse) [inline src/Command/Argument/Hash/HSetFArguments.php:81]: Check that at most one modifier from the group is used.

File: src/Command/Argument/Hash/HSetFArguments.php
Comment (gerzse) [inline src/Command/Argument/Hash/HSetFArguments.php:94]: Same like for HGETF, what if someone first enables `keepTtl` and then calls `setTtlModifierEnum`? There should be a check on the other side too?

File: src/Command/Redis/HEXPIRETIME.php
Comment (gerzse) [inline src/Command/Redis/HEXPIRETIME.php:17]: I think this one too has the {NX,XX,LT,GT} flags. Maybe it should extend `HEXPIRE`?

File: src/Connection/Replication/MasterSlaveReplication.php
Comment (gerzse) [inline src/Connection/Replication/MasterSlaveReplication.php:75]: An idea: put these changes not strictly related to hash field expiration in a separate PR, to avoid noise?

File: src/ClientContextInterface.php
Comment (vladvildanov) [reply src/ClientContextInterface.php:178]: Makes sense

File: src/Command/Argument/Hash/CommonExpiration.php
Comment (vladvildanov) [reply src/Command/Argument/Hash/CommonExpiration.php:42]: Got it

File: src/Command/Argument/Hash/CommonExpiration.php
Comment (vladvildanov) [reply src/Command/Argument/Hash/CommonExpiration.php:23]: So something like `overrideModifierEnum` sounds better?

File: src/Command/Argument/Hash/HGetFArguments.php
Comment (vladvildanov) [reply src/Command/Argument/Hash/HGetFArguments.php:26]: Okay, let's add the same kind of check on the other side

File: src/Connection/Replication/MasterSlaveReplication.php
Comment (vladvildanov) [reply src/Connection/Replication/MasterSlaveReplication.php:75]: There's already a PR with this changes, I'm waiting for approval and then rebase this branch

File: src/Command/Redis/HEXPIRETIME.php
Comment (vladvildanov) [reply src/Command/Redis/HEXPIRETIME.php:17]: @gerzse This is not reflected in the PRD, I think in this case we need to let it know for others as well

Comment (sazzad16) [reply src/Command/Redis/HEXPIRETIME.php:17]: @gerzse No. HEXPIRETIME and similar commands are **read** commands. These don't have {NX,XX,LT,GT} flags.

File: src/Command/Argument/Hash/CommonExpiration.php
Comment (gerzse) [inline src/Command/Argument/Hash/CommonExpiration.php:92]: Should this be two strings separated by comma, instead of one string?

**Review by gerzse — COMMENTED** [review]
Looks good to me. I can't use the "Approve" button, but consider this approved from my side.

File: src/Command/Argument/Hash/CommonExpiration.php
Comment (vladvildanov) [reply src/Command/Argument/Hash/CommonExpiration.php:92]: Wow, that's a goddamn typo, thank you

**Review by tillkruss — APPROVED** [review]

**Comment by jdecool** [comment]
Very interesting.

Can it be merged soon ?

**Comment by rusowyler** [comment]
Hey guys! Can we merge this PR soon?

**Pull request merged**
